funnel/views/commentvote.py

```diff
@@ -127,6 +131,60 @@ def parent_comments_url(obj):
     return url
 
 
+@route('/comments')
+class AllCommentsView(ClassView):
+    current_section = 'comments'
+
+    @route('', endpoint='comments')
+    @requires_login
+    @render_with(json=True)
+    def view(self):
+        commentset_memberships = [
+            {
+                'parent': cm.commentset.parent.current_access(
+                    datasets=('primary', 'related')
+                ),
+                'commentset_url': cm.commentset.url_for(_external=True),
+                'last_seen_at': cm.last_seen_at,
+                'new_comments_count': (
+                    cm.commentset.comments.filter(Comment.state.PUBLIC)
+                    .filter(Comment.created_at > cm.last_seen_at)
+                    .count()
+                ),
+                'last_comment': (
+                    cm.commentset.comments.filter(Comment.state.PUBLIC)
+                    .order_by(Comment.created_at.desc())
+                    .first()
+                    .current_access(datasets=('primary', 'related'))
+                ),
+            }
+            for cm in (
+                current_auth.user.active_commentset_memberships.join(Commentset)
+                .join(Comment)
+                .filter(Comment.state.PUBLIC)
+                .order_by(Comment.created_at.desc())
```

Comment:
This join for the sake of ordering is going to be a performance bottleneck. Since a new comment affects all subscribers of the commentset, a `last_comment_at` timestamp should be added on the Commentset model, and should be updated by the view when a comment is posted (+initially populated in a migration). This will remove the join on Comment on this query.